### ISSUE TITLE
Set pump time

### ISSUE THREAD

**Issue by Tausta** [pazaan/600SeriesAndroidUploader#151]
Nowadays uploader checks pump time and compare it to phone time. Then this time offset has been used to manipulate pump records timestamp. But with medical devices records, typically communication devises do not manipulate records.

Better solution:
 -  uploader checks pump time, and if offset is more than 5 (?) seconds uploader set pump time
 - the user must give permission to set the pump time
 - uploader do not manipulate records timestamp

Especially if in the future pump history records are transferred to NS, it is important that uploader is as transparent as possible.

**Comment by volkerrichert** [comment]
Is there a way to set the pump time remotely? I can't remember someone.

**Comment by pazaan** [comment]
There is a protocol message for setting the time. I haven't documented it yet.

**Comment by volkerrichert** [comment]
Ah, ok. Cool. Maybe we should add an option to enable sync time and not just change it.

**Comment by Pogman** [comment]
I added a warning message and a help message that's in v5.1 of the uploader for this. If the time difference is over 10 minutes the user gets this warning. 

I don't think it's a good idea to change anything on the pump via the uploader. At least for something that's being released to the general public.

**Comment by volkerrichert** [comment]
me, too. Reading is fine, but writing... So maybe an option or a message an an upload to NS as a message

**Comment by Tausta** [comment]
Yep, of course this kind of feature have to be by default off.  But option that user can activate. I also think that by default uploader should not manipulate history records timestamps.

**Comment by Tausta** [comment]
Sorry, wrong button. Reopen.

**Comment by volkerrichert** [comment]
This may cause issues finding old records.

**Comment by Pogman** [comment]
interesting point... that could be why the pump uses 2 time data points, the RTC and the OFFSET.

**Comment by pazaan** [comment]
It is indeed. Changing pump date/time changes offset, not RTC. 
Agree with not writing data to the pump.

**Comment by Pogman** [comment]
Things can get messy if the time on the uploader device is changed. The new treatment stuff I've been working on relies on good uploader records and time moving forward. Need to think a bit on this type of situation....

**Comment by Tausta** [comment]
I fully agree that writing data to the pump must be done with special care and avoiding risks. If automatic time sync is too big step, maybe manual time sync can be first step. Can in uploader menu be "Sync pump time", and when user activate it uploader tries to set pump time.

**Comment by volkerrichert** [comment]
The first step should be a warming  and maybe an icon

**Comment by Pogman** [comment]
The warning is already in there:

https://github.com/pazaan/600SeriesAndroidUploader/blob/d582d467fa67d9062a1227e3e2ba8b77068f4d83/app/src/main/java/info/nightscout/android/medtronic/service/MedtronicCnlIntentService.java#L293
